devicetypes/smartthings/ct100-thermostat.src/ct100-thermostat.groovy

```diff
@@ -636,7 +637,8 @@ def switchMode() {
 
 def switchToMode(nextMode) {
 	def supportedModes = state.supportedModes
-	if (supportedModes) {
+	// Old version of supportedModes was as string, make sure it gets updated
+	if (supportedModes && supportedModes[0].size() > 1) {
```

Comment:
Same question about supportedModes containing an empty list.  It's a bit obsessive, but you still might want to check if supportedModes is null and it's size > 0